rules/sql.go

```diff
@@ -51,10 +51,17 @@ func (s *sqlStrConcat) ID() string {
 }
 
 // see if we can figure out what it is
-func (s *sqlStrConcat) checkObject(n *ast.Ident) bool {
+func (s *sqlStrConcat) checkObject(n *ast.Ident, c *gosec.Context) bool {
 	if n.Obj != nil {
 		return n.Obj.Kind != ast.Var && n.Obj.Kind != ast.Fun
 	}
+
```

Comment:
It would be nice to have a test for this use case.

Reply:
👍 I was thinking so too. But this test case spans across two files and I'm not sure how to do it.

Reply:
I think you can extend the `sample.Code` to support multiple files. The test package is built here: https://github.com/securego/gosec/blob/5f98926a7be81fd2fcb25b535e2b4e30406a8af0/rules/rules_test.go#L35

In this file are the code samples defined:
https://github.com/securego/gosec/blob/master/testutils/source.go

Reply:
Looks fine. You could have also use a `Code []string`. To keep multiple files. I will refactoring afterwards.

Thanks for providing the tests.

Reply:
ahh yes. It's much better to use `Code []string`. If you are okay, can I help with the refactoring? for https://hacktoberfest.digitalocean.com/ 😂

Reply:
I refactored it in the PR https://github.com/securego/gosec/pull/248